src/containers/store/actions/creators/AppConfig.js

```diff
@@ -174,4 +255,61 @@ export function setQRData(data) {
   return (dispatch) => {
     dispatch({type: actionType.SAVE_QR_SCANNER_DATA, payload: data});
   }
+}
+
+// balance calculating function brought from fetchCoinData file for dispatch combination
+function calculateBalances(tokenQuantities, tokenConversionMatrix) {
```

Comment:
This is a nasty for loop... 
@priom Anyway to optimize i wonder?

Reply:
could we add a commnet showing us what the params look like?

Reply:
I couldn't understand fully how the function worked, so I left it as it was.

Reply:
Optimization. If its been < x hours (maybe 2-5?) don't refresh the currencies. Prices won't change that much (on the fiat side(